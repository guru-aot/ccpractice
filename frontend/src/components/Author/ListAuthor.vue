

  /**
   * ListViewComponent Page
   */
  
  <template>
  <v-card class="mx-auto">
    <v-toolbar flat>
      <v-toolbar-title class="grey--text">All Author</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-card-text v-if="getAuthorList && getAuthorList.length == 0">No Author found. Add one</v-card-text>
    <v-card-text v-if="getAuthorList && getAuthorList.length != 0">
      <v-simple-table fixed-header>
        <template v-slot:default>
          <thead>
            <tr>
              <th class="text-left">SI</th>
              <th class="text-left">first_name</th>
              <th class="text-left">last_name</th>
              <th class="text-left">middle_name</th>
            </tr>
          </thead>
          <tbody>
            <tr v-for="(author ,idx) in  getAuthorList" :key="author.id">
              <td>{{ idx+1 }}</td>
              <td>{{ author.first_name }}</td>
              <td>{{ author.last_name }}</td>
              <td>{{ author.middle_name }}</td>
            </tr>
          </tbody>
        </template>
      </v-simple-table>
    </v-card-text>
  </v-card>
</template>
  <script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
const AuthorModule = namespace('AuthorModule');

@Component
export default class ListAuthorComponent extends Vue {
  @AuthorModule.Action('loadAuthor') public loadAuthor!: any;
  @AuthorModule.Getter('getAuthorList') public getAuthorList!: [];

  private mounted() {
    this.loadAuthor();
  }
}
</script>
 