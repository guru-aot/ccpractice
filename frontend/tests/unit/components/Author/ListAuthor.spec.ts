/**
 * ListAuthor test
 */

import { shallowMount, createLocalVue } from '@vue/test-utils';
import ListAuthor from '@/components/Author/ListAuthor.vue';

import Vuex from 'vuex';
import vuetify from 'vuetify';

describe('component/Author/ListAuthor.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(vuetify);
    const store = new Vuex.Store({
      modules: {
        AuthorModule: {
          namespaced: true,
          state: {},
          getters: {
            getAuthorList: jest.fn(() => [
              {
                created: '2019-10-10T05:59:56.999852',
                first_name: 'author1 ',
                id: 1,
                last_name: 'last',
                middle_name: '',
                modified: '2019-10-10T05:59:56.999881',
              },
            ]),
            successStatus: jest.fn(),
            errorStatus: jest.fn(),
          },
          actions: {},
        },
      },
    });
    wrapper = shallowMount(ListAuthor, {
      localVue,
      store,
    });
  });
  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
