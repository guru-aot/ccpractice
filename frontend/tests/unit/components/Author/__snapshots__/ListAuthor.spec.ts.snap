// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Author/ListAuthor.vue renders props when passed 1`] = `
<v-card-stub
  class="mx-auto"
  loaderheight="4"
  tag="div"
>
  <v-toolbar-stub
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    tile="true"
  >
    <v-toolbar-title-stub
      class="grey--text"
    >
      All Author
    </v-toolbar-title-stub>
     
    <div
      class="flex-grow-1"
    />
  </v-toolbar-stub>
   
  <v-divider-stub />
   
  <!---->
   
  <v-card-text-stub>
    <v-simple-table-stub
      fixedheader="true"
    />
  </v-card-text-stub>
</v-card-stub>
`;
